content/articles/Dependency-injection-in-flutter/index.md

```diff
@@ -0,0 +1,157 @@
+### Introduction 
+`Dependency injection` is an object-oriented technique that sends the dependencies of another object to an object. Using dependency injection, we can also move the creation and restriction of dependent objects outside the classes. This brings a more significant level of adaptability, decoupling, and simpler testing.
+
+The `provider package` wraps Inherited Widgets in an easy-to-use wrapper. It makes it easy to manage and utilize, and it also includes a state management mechanism for managing data within the app.
+
+In this article, we will learn how to implement dependency injection, get rid of flutter packages, and the reasons why we need dependency injection in a flutter.
+
+### Table of content
+- [Dependency injection in Flutter](#dependency-injection-in-Flutter)
+- [Functionalities of dependency injection](#functionalities-of-dependency-injection)
+- [Reason why we need dependency injection](#Reason-why-we-need-dependency-injection)
+- [How to get rid of flutter package](#How-to-get-rid-of-flutter-package)
+- 
+### Dependency injection in Flutter
+Although dependency injection is a simple example, libraries are frequently used to abstract it from the designer. Many of these libraries take advantage of reflection (mirrors in Dart). There are issues in a flutter:
+1. Mirrors are crippled for execution reasons. 
+2. The settled idea of widgets makes it unfeasible to pass conditions many levels down the tree.
+
+The inject library, on the other hand, can be used to solve similar issues.
+Annotations used by the inject library are listed below.
+
+1. **@Injector**-An inversion of control container developed from a bunch of modules.
+2. **@provides and @module**-define classes and methods providing dependencies.
+3. **@component**-used in performing an injection.
+
+### Functionalities of dependency injection
+**Concrete class injection**
+
+Can be like below:
+
+```
+import ‘package:inject/inject.dart’;
+@provide
+class ExampleExa {
+   // implementation
+}
+```
+
+**Interface injection**
+
+First, create an abstract class then provide dependencies in the module.
+Abstract class:
+
+```
+abstract class NewExa{
+  
+}class OldExa implements NewExa {
+   @override
+    {
+   // implementation
+   }
+}
+```
+
+Final module:
+
+```
+import ‘package:inject/inject.dart’;
+@module
+class ExampleExa {
+@provide
+NewExa newExa() => OldExa();
+}
+```
+
+**Providers**

```

Comment:
Kindly add more content here to make the functionalities of DI  more understandable. I feel like it's just code with no clear explanation. Besides, a beginner would not know what you are trying to achieve.